Source/WebGPU/WebGPU/Pipeline.mm

```diff
@@ -183,20 +183,25 @@ bool validateBindGroup(BindGroup& bindGroup)
                 continue;
 
             auto it = bindGroupLayoutEntries.find(bindingIndex);
-            RELEASE_ASSERT(it != bindGroupLayoutEntries.end());
-
-            auto* bufferBinding = get_if<WGPUBufferBindingLayout>(&it->value.bindingLayout);
-            if (!bufferBinding)
-                continue;
+            if (it == bindGroupLayoutEntries.end())
+                return [NSString stringWithFormat:@"Buffer size is missing for binding at index %u bind group", bindingIndex];
+
+            uint64_t bufferSize = 0;
+            if (bufferSizes) {
```

Comment:
I don't think this logic is correct. I think the user-provided value takes precedence over the computed value, but we're still missing the check that the user-provided value is not smaller than the computed value.